### ISSUE TITLE
TFT: Custom multi-model loss function

### ISSUE THREAD

**Issue by VOIDSTobias** [jdb78/pytorch-forecasting#785]
Hi,

not really an issue, but rather a question from an applied business context: Would it be possible to implement a custom loss function with business context, e.g. a tweedie loss (from M5 Competition, link below), and combine this with a multi-model distribution such as in the paper here: https://arxiv.org/abs/1803.03800?

<img width="923" alt="Screenshot 2021-12-05 at 21 28 57" src="https://user-images.githubusercontent.com/79939008/144764726-73b7bde3-7076-4f3c-aa92-9bc55b1b06a1.png">

Do you believe this makes sense and could improve value for a company?

I really would be interested in how to pull this off! Thanks.

M5 Link: https://medium.com/artefact-engineering-and-data-science/sales-forecasting-in-retail-what-we-learned-from-the-m5-competition-445c5911e2f6

**Comment by Yashasvikhatri9** [comment]
Hi @VOIDSTobias,

I am new to contributing to this repository and I found this discussion very interesting.

From what I understand, implementing a custom loss such as Tweedie could be done by creating a new class that inherits from `MultiHorizonMetric` and defining the loss calculation there. The output generation might then need adjustment in `FullyConnectedMultiOutputModule` to support the desired distribution.

I would like to explore this further and potentially work on an example implementation if this is still relevant to the project.

Could a maintainer confirm whether this approach would be appropriate?

Thanks!
